opencti-platform/opencti-graphql/src/utils/safeEjs.client.ts

```diff
@@ -0,0 +1,157 @@
+import { Worker } from 'worker_threads';
+import * as path from 'path';
+import type { Data } from 'ejs';
+import type { SafeRenderOptions } from './safeEjs';
+import type { WorkerReply } from './safeEjs.worker';
+
+export type WorkerOptions = {
+  timeout?: number,
+  useJsonEscape?: boolean,
+  resourceLimits?: {
+    maxOldGenerationSizeMb?: number
+    maxYoungGenerationSizeMb: number,
+    codeRangeSizeMb: number,
+    stackSizeMb: number,
+  }
+};
+
+// Helper to serialize functions and validate data
+const prepareDataForWorker = (data: Data): Data => {
+  const forbidden = ['eval', 'Function', 'constructor', '__proto__', 'prototype', 'freeze'];
+
+  // Track function results to create a wrapper object
+  const functionMap = new Map<string, any>();
+  let functionCounter = 0;
+
+  const serialize = (obj: any, currentPath: string = ''): any => {
+    if (obj === null || obj === undefined) return obj;
+
+    if (typeof obj === 'function') {
+      // Functions can't be passed to workers, so we create a special marker
+      // and store the function's result
+      functionCounter += 1;
+      const funcId = `__func_${functionCounter}__`;
+      try {
+        functionMap.set(funcId, obj());
```

Comment:
how can it work without arguments ?
we should add a test case with the usage of markdownToHtml